docs/developer-docs/latest/setup-deployment-guides/configurations/required/server.md

```diff
@@ -0,0 +1,115 @@
+---
+title: Server configuration - Strapi Developer Documentation
+description: 
+---
+
+<!-- TODO: update SEO -->
+
+# Server configuration
+
+The `./config/server.js` is used to define server configuration for the Strapi application.
+
+:::: tabs card
+
+::: tab Minimal
+
+## Minimal Server Config
+
+This is the default config created with any new project, all these keys are required at the very least, environmental configs do not need to contain all these values so long as they exist in the default `./config/server.js`.
+
+**Path —** `./config/server.js`.
+
+```js
+module.exports = ({ env }) => ({
+  host: env('HOST', '0.0.0.0'),
+  port: env.int('PORT', 1337),
+  admin: {
+    auth: {
+      secret: env('ADMIN_JWT_SECRET', 'someSecretKey'),
+    },
+  },
+});
+```
+
+:::
+
+::: tab Full
+
+## Full Server Config
+
+This is an example of a full configuration, typically certain keys do not need to present in environmental configs, and not all of these keys are required. Please see the table below to see what each key does.
+
+**Path —** `./config/server.js`.
+
+```javascript
+module.exports = ({ env }) => ({
+  host: env('HOST', '0.0.0.0'),
+  port: env.int('PORT', 1337),
+  socket: '/tmp/nginx.socket', // only use if absolutely required
+  emitErrors: false,
+  url: env('PUBLIC_URL', 'https://api.example.com'),
+  proxy: env.bool('IS_PROXIED', true),
+  cron: {
+    enabled: env.bool('CRON_ENABLED', false),
+  },
+  admin: {
+    auth: {
+      events: {
+        onConnectionSuccess(e) {
+          console.log(e.user, e.provider);
+        },
+        onConnectionError(e) {
+          console.error(e.error, e.provider);
+        },
+      },
+      secret: env('ADMIN_JWT_SECRET', 'someSecretKey'),
+    },
+    url: env('PUBLIC_ADMIN_URL', '/dashboard'),
+    autoOpen: false,
+    watchIgnoreFiles: [
+      './my-custom-folder', // Folder
+      './scripts/someScript.sh', // File
+    ],
+    host: 'localhost', // Only used for --watch-admin
+    port: 8003, // Only used for --watch-admin
+    serveAdminPanel: env.bool('SERVE_ADMIN', true),
+    forgotPassword: {
+      from: 'no-reply@example.com',
+      replyTo: 'no-reply@example.com',
+    },
+  },
+});
+```
+
+:::
+
+::::
+
+### Available options
```

Comment:
Can you put a TODO in here for `add admin jwt config option`

I know what it refers to and I can do a PR to add it but I don't want to worry about small stuff til we get this merged in and I'll do it later.

Reply:
No (albeit that also needs to be handled too). I'm referring to this: https://github.com/derrickmehaffy/strapi-example-admin-expire-date

Which has existed for a while but was never documented.

Reply:
(I added a comment to that notion doc since I don't really agree with how it's planned to be handled)

Reply:
OK! 
✅ TODO [added](https://github.com/strapi/documentation/pull/429/commits/0e0a3e03bc54fcddb8eebebd0b64e26b2efd4269) :-)